### PR TITLE
SALTO-2349: Add enableJSM to user fetch config

### PR DESCRIPTION
**Pull request by idoamit8**
In this PR I added the flag enableJSM to user Fetch config. 
---

None

---
_Release Notes_: 
None

---
_User Notifications_: 
None

### CODE REVIEW

**Review by IdoZakk — COMMENTED** [review]
The config creator is for the SAAS UI, I do not think we do it for now

File: packages/jira-adapter/src/config_creator.ts [hunk 1]
```diff
@@ -49,6 +51,9 @@ export const getConfig = async (
   if (options !== undefined && isOptionsTypeInstance(options) && options.value.enableScriptRunnerAddon) {
     defaultConf.value.fetch.enableScriptRunnerAddon = true
   }
+  if (options !== undefined && isOptionsTypeInstance(options) && options.value.enableJSM) {
+    defaultConf.value.fetch.enableJSM = true
+  }
```
Comment (omrifarkash) [inline packages/jira-adapter/src/config_creator.ts:56]:
```suggestion
  if (options !== undefined && isOptionsTypeInstance(options)) {
    if (options.value.enableScriptRunnerAddon) {
       defaultConf.value.fetch.enableScriptRunnerAddon = true
    }
    if (options.value.enableJSM) {
       defaultConf.value.fetch.enableJSM = true
    }
  }
```

**Review by omrifarkash — COMMENTED** [review]
Looks good, few small comments

File: packages/jira-adapter/test/config_creator.test.ts [hunk 2]
```diff
@@ -39,22 +42,31 @@ describe('config creator', () => {
...
+      }
     )
     const config = await getConfig(options)
     expect(config).toEqual(await createDefaultInstanceFromType(ElemID.CONFIG_NAME, configType))
   })
-  it('get config should return return correctly when true', async () => {
+  it('get config should return correctly when enableScriptRunnerAddon is true', async () => {
     const config = await getConfig(options)
     expect(config.value.fetch.enableScriptRunnerAddon).toBeTrue()
+    expect(config.value.fetch.enableJSM).toBeTrue()
```
Comment (omrifarkash) [inline packages/jira-adapter/test/config_creator.test.ts:66]: Please remove this assertion from the ScriptRunner `it`
It is just confusing things. Maybe you can also set enableJSM to false in this `it` and do the reverse thing in the next one (this is a nit)

**Review by omrifarkash — APPROVED** [review]
Approved with small comment

**Pull request merged**